### DISCUSSION TITLE
Difference between exceptions thrown by SESv2 `SendEmail`

### DISCUSSION THREAD

**Discussion by lonix1** [aws/aws-sdk-net#3510] [Q&A] [answered]
When using `SendEmailAsync` in the SESv2 client, it can return these exceptions (inter alia):
- [TooManyRequestsException](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/SimpleEmailV2/TTooManyRequestsException.html) - "Too many requests have been made to the operation"
- [LimitExceededException](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/SimpleEmailV2/TLimitExceededException.html) - "There are too many instances of the specified resource type"

How do those exceptions differ? And what does "resource type" mean in this context?

I tested by sending many requests in parallel, and received a `TooManyRequestsException`. I've not seen the other exception.

**Comment by bhoradc** [comment, marked as answer]
Hi @lonix1, 

Thank you for reaching us with your query. 

`TooManyRequestsException` is raised when you have exceeded the rate limit for the specific operation you're calling, in this case its the `SendEmailAsync`. 

`LimitExceededException`  is raised when you exceed a limit or quota for a specific resource type within the SES service. The `resource type` could refer to various entities, such as Recipient limits, Email sending quotas, etc. 

The [Service quotas in Amazon SES](https://docs.aws.amazon.com/ses/latest/dg/quotas.html) lists and describes the quotas that apply to Amazon SES resources and operations. Exceeding these specific resource types and limits would cause the `LimitExceededException` in SES.

Let me know if this helps. 

Regards, 
Chaitanya

**Reply by lonix1** [reply]
Thank you, makes sense now!

Pity that info isn't in the docs, it's very helpful.
